source/common/ssl/connection_impl.cc

```diff
@@ -211,6 +211,7 @@ Network::ConnectionImpl::IoResult ConnectionImpl::doWriteToSocket() {
     // at the beginning of the buffer.
     if (inner_bytes_written > 0) {
       write_buffer_->drain(inner_bytes_written);
+      checkForLowWatermark();
```

Comment:
Instead of doing this inside both Network::ConnectionImpl and Ssl::ConnectionImpl is it possible to do this in the common code that calls doWriteToSocket() ? Just a thought.